src/main/java/scmspain/karyon/restrouter/RouteHandlerProvider.java

```diff
@@ -0,0 +1,38 @@
+package scmspain.karyon.restrouter;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+import com.netflix.config.DynamicPropertyFactory;
+import com.netflix.hystrix.contrib.rxnetty.metricsstream.HystrixMetricsStreamHandler;
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.RequestHandler;
+
+public class RouteHandlerProvider implements Provider<RequestHandler<ByteBuf, ByteBuf>> {
```

Comment:
Yes, I don't know how to name this too.

Maybe if this provider instantiate the RestRouterHandler the name of this provider would make more sense, imo.

If there isn't an idea on how to name this better, maybe it's ok with this one..